### ISSUE TITLE
permalink: false not working for a *.11ty.js

### ISSUE THREAD

**Issue by thiagomgd** [11ty/eleventy#2260]
**Describe the bug**
On my `_src/notion_posts/notion_post.11ty.js` I use `excludeFromCollections` + `permalink: false` to not render draft posts. Only on this file, it causes the following error:
`[11ty] > The "path" argument must be of type string. Received type boolean (false)`

Those posts are loaded from `_data/notion_posts.js`.

The same draft logic works on `_src/posts/posts.11tydata.js`, which processes the markdown templates on that folder.

**To Reproduce**

1. Clone my 'buildfail' branch - https://github.com/thiagomgd/geekosaurblog/tree/buildfail 
2. `npm run build` or `npm run start:prod` 

**Expected behavior**
Build without error, draft posts are not created

**Environment:**

- OS and Version: Mac
- Eleventy Version: 1.0.0

**Comment by thiagomgd** [comment]
For now, I made it "work" by omitting the draft posts from the posts collection. The downside is that the pages will still exist, even if they are not referenced anywhere:
```
eleventyConfig.addCollection('posts', collection => {
    return collection.getFilteredByTag('post')
      .filter(post => post.data.eleventyExcludeFromCollections !== true)
      .sort(function(a, b) {
        const timeA = a.data.created_date ? a.data.created_date.getTime() : 0;
        const timeB = b.data.created_date ? b.data.created_date.getTime() : 0;
        // console.log(a.data.title, b.data.title, timeA, timeB);
        return timeA - timeB;
      });
  });
```

**Comment by pdehaan** [comment]
Interesting, it seems to work for me in a bare repo: https://github.com/pdehaan/11ty-2260

I created three files:
- src/index.11ty.js
	- `permalink: "/index-page/"`
- src/draft.11ty.js
	- `permalink: false`
	- `eleventyExcludeFromCollections: true`
- src/archive.11ty.js
	- `eleventyExcludeFromCollections: true`

Only my www/index-page/index.html and www/archive/index.html files are written, and the archive page only lists `/index-page/` since both the draft and archive page itself are excluded from collections.

**Comment by pdehaan** [comment]
Not sure I can run your repo, since I'll probably be missing [at least] a `process.env.NOTION_API_KEY`.

**Comment by thiagomgd** [comment]
@pdehaan 

> Interesting, it seems to work for me in a bare repo

that's weird. Maybe one of the plugins is affecting it in some way?

> Not sure I can run your repo.

You can, it's not going to break. I have a check for the key on the data files that call the notion API, and they are just going to skip fetching more content. `npm run start:prop` will not provide the token and will just load the results from the cache files.

**Comment by thiagomgd** [comment]
Got my dependencies down to only those, and still have the problem, so it's probably not that:

```
"devDependencies": {
    "@11ty/eleventy": "^1.0.0",
    "@notionhq/client": "^0.4.13",
    "cheerio": "^1.0.0-rc.10",
    "lodash": "^4.17.21",
    "luxon": "^1.26.0",
    "markdown-it": "^12.3.2",
    "node-fetch": "^2.6.7",
    "notion-to-md": "^2.3.0",
    "outdent": "^0.8.0"
  }
```

Regarding it working for you on a basic repo, maybe it only happens when you have pagination? Or pagination on an async data file like my `_data/notion_posts.js`?

**Comment by pdehaan** [comment]
I think you're correct! It does seem pagination related (and possibly *.11ty.js specific):

```js
// src/pagination.11ty.js
class Pagination {
  get data() {
    return {
      pagination: {
        data: "notion_posts",
        size: 1,
        alias: "notion_post",
        addAllPagesToCollections: true,
        // before(data) {
        //   return data.filter(item => item.permalink !== false);
        // },
      },
      eleventyComputed: {
        permalink(data) {
          return data.notion_post?.permalink;
        },
        eleventyExcludeFromCollections(data) {
          return data.notion_post?.eleventyExcludeFromCollections || false;
        }
      }
    };
  }
  render(data) {
    return `title=${data.notion_post.title}`;
  }
};

module.exports = Pagination;
```

```js
// src/_data/notion_post.js
module.exports = [
  {
    permalink: false,
    eleventyExcludeFromCollections: true,
    title: "permalink=false, eleventyExcludedFromCollections:true",
  },
  {
    permalink: false,
    title: "permalink=false",
  },
  {
    permalink: "paginated-one/",
    eleventyExcludeFromCollections: true,
    title: "permalink=paginated-one/, eleventyExcludedFromCollections:true",
  },
  {
    permalink: "paginated-two/",
    eleventyExcludeFromCollections: false,
    title: "permalink=paginated-two/, eleventyExcludedFromCollections:false",
  }
];
```

### OUTPUT
```sh
> eleventy

[11ty] Writing www/archive/index.html from ./src/archive.11ty.js
[11ty] Writing www/index-page/index.html from ./src/index.11ty.js
[11ty] Writing false from ./src/pagination.11ty.js
[11ty] Writing false from ./src/pagination.11ty.js
[11ty] Writing www/paginated-one/index.html from ./src/pagination.11ty.js
[11ty] Writing www/paginated-two/index.html from ./src/pagination.11ty.js
[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] > Having trouble writing template: www/paginated-one/index.html

`TemplateWriterWriteError` was thrown
[11ty] > The "path" argument must be of type string. Received type boolean (false)

`TypeError` was thrown:
[11ty]     TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type boolean (false)
        at new NodeError (node:internal/errors:371:5)
        at validateString (node:internal/validators:120:11)
        at Object.parse (node:path:1450:5)
        at Template._write (/private/tmp/11ty-2260/node_modules/@11ty/eleventy/src/Template.js:817:36)
        at /private/tmp/11ty-2260/node_modules/@11ty/eleventy/src/Template.js:923:23
        at async Promise.all (index 0)
        at async Promise.all (index 4)
        at async Eleventy.executeBuild (/private/tmp/11ty-2260/node_modules/@11ty/eleventy/src/Eleventy.js:986:13)
[11ty] Wrote 0 files in 0.03 seconds (v1.0.0)
```

Workaround seems to be to filter `permalink=false` items out in [the `before()` callback](https://www.11ty.dev/docs/pagination/#the-before-callback).

**Comment by thiagomgd** [comment]
Thanks for the workaround suggestion!

**Comment by zachleat** [comment]
Howdy y’all I believe this is a duplicate of #2167, already shipped in 2.0.0-canary.6!
